src/main/webapp/player/simid_player.html

```diff
@@ -65,6 +64,41 @@
       }
       document.getElementById('ad_params').value = adParams;
     }
+
+    window.getLocationPermission = () => {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function (position) {
+          const pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
+          };
+          let adParamObject = document.getElementById('ad_params').value;
+          let adParams = JSON.parse(adParamObject);
+          adParams.userCoordinates = pos;
+          adParamObject = JSON.stringify(adParams);
+          document.getElementById('ad_params').value = adParamObject;
+        }, (error) => { //callback function that takes a GeolocationPositionError 
+          handleLocationError(error, true);
+        });
+      } else {
+        // Browser doesn't support Geolocation
+        handleLocationError(error, false);
+      }
+    }
+
+    window.handleLocationError = (error, geolocationAllowed) => {
```

Comment:
Since this function isn't being used outside of this module, I don't think youll need to put this one into the window